flang/lib/Optimizer/Transforms/OMPMapInfoFinalization.cpp

```diff
@@ -59,6 +59,34 @@ class OMPMapInfoFinalizationPass
            /*corresponding local alloca=*/fir::AllocaOp>
       localBoxAllocas;
 
+  unsigned long getDescriptorMapType(unsigned long mapTypeFlag,
+                                     mlir::Operation *target) {
+    auto newDescFlag = llvm::omp::OpenMPOffloadMappingFlags(mapTypeFlag);
```

Comment:
Nit: To shorten the `isa_and_nonnull` to `isa` checks below and improve readability a bit, maybe add here:
```c++
  if (!target)
    return mapTypeFlag;
```